armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -408,19 +411,30 @@ def _getAverageComponentTemperature(self, compIndex):
         weights = numpy.array([self.getWeight(b) / b.getHeight() for b in blocks])
         weights /= weights.sum()  # normalize by total weight
         components = [sorted(b.getComponents())[compIndex] for b in blocks]
-        avgComponentMass = sum(c.getMass() for c in components) / len(components)
-        if avgComponentMass == 0.0:
+        weightedAvgComponentMass = sum(
+            w * c.getMass() for w, c in zip(weights, components)
+        )
+        if weightedAvgComponentMass == 0.0:
             # if there is no component mass (e.g., gap), do a regular average
             return numpy.mean(numpy.array([c.temperatureInC for c in components]))
         else:
             return (
                 weights.dot(
                     numpy.array([c.temperatureInC * c.getMass() for c in components])
                 )
-                / avgComponentMass
+                / weightedAvgComponentMass
             )
 
-    def _checkBlockSimilarity(self):
+    def _averageByComponent(self):
```

Comment:
I think it's more clear to leave this as `_checkBlockSimilarity`. Having a method called `_averageByComponent` and an attribute called `averageByComponent` is a little confusing. Plus this methods functionality remains mostly unchanged so leaving its name makes sense to me.

Reply:
I thought "_checkBlockSimilarity" didn't fully encompass what it was doing since it now checks `self.averageByComponent`

Reply:
I broke it out into a separate function